src/rpcmasternode.cpp

```diff
@@ -601,5 +608,222 @@ UniValue masternodelist(const UniValue& params, bool fHelp)
+            CMasternodeBroadcast mnb;
+
+            bool result = activeMasternode.CreateBroadcast(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), errorMessage, mnb, true);
+
+            UniValue statusObj(UniValue::VOBJ);
+            statusObj.push_back(Pair("alias", mne.getAlias()));
+            statusObj.push_back(Pair("result", result ? "successful" : "failed"));
+
+            if(result) {
+                successful++;
+                CDataStream ssMnb(SER_NETWORK, PROTOCOL_VERSION);
+                ssMnb << mnb;
+                statusObj.push_back(Pair("hex", HexStr(ssMnb.begin(), ssMnb.end())));
+            } else {
+                failed++;
+                statusObj.push_back(Pair("errorMessage", errorMessage));
+            }
+
+            resultsObj.push_back(Pair("status", statusObj));
+        }
+        pwalletMain->Lock();
+
+        UniValue returnObj(UniValue::VOBJ);
+        returnObj.push_back(Pair("overall", strprintf("Successfully created broadcast messages for %d masternodes, failed to create %d, total %d", successful, failed, successful + failed)));
+        returnObj.push_back(Pair("detail", resultsObj));
+
+        return returnObj;
+    }
+
+    if (strCommand == "decode")
+    {
+        if (params.size() != 2)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Correct usage is 'masternodebroadcast decode \"hexstring\"'");
+
+        CMasternodeBroadcast mnb;
+
+        if (!DecodeHexMnb(mnb, params[1].get_str()))
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Masternode broadcast message decode failed");
+
+        if(!mnb.VerifySignature())
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Masternode broadcast signature verification failed");
+
+        UniValue resultObj(UniValue::VOBJ);
+
+        resultObj.push_back(Pair("vin", mnb.vin.ToString()));
+        resultObj.push_back(Pair("addr", mnb.addr.ToString()));
+        resultObj.push_back(Pair("pubkey", CBitcoinAddress(mnb.pubkey.GetID()).ToString()));
+        resultObj.push_back(Pair("pubkey2", CBitcoinAddress(mnb.pubkey2.GetID()).ToString()));
+        resultObj.push_back(Pair("vchSig", EncodeBase64(&mnb.vchSig[0], mnb.vchSig.size())));
+        resultObj.push_back(Pair("sigTime", mnb.sigTime));
+        resultObj.push_back(Pair("protocolVersion", mnb.protocolVersion));
+        resultObj.push_back(Pair("nLastDsq", mnb.nLastDsq));
+
+        UniValue lastPingObj(UniValue::VOBJ);
+        lastPingObj.push_back(Pair("vin", mnb.lastPing.vin.ToString()));
+        lastPingObj.push_back(Pair("blockHash", mnb.lastPing.blockHash.ToString()));
+        lastPingObj.push_back(Pair("sigTime", mnb.lastPing.sigTime));
+        lastPingObj.push_back(Pair("vchSig", EncodeBase64(&mnb.lastPing.vchSig[0], mnb.lastPing.vchSig.size())));
+
+        resultObj.push_back(Pair("lastPing", lastPingObj));
+
+        return resultObj;
+    }
+
+    if (strCommand == "relay")
```

Comment:
You mean create an array (vector?) and serialize it in `create-all`, right? And then update `decode` and `relay` to the same format? I think it's a good idea and can be improved this way imo. One thing I'm not sure about is will the size of it for say 10+ masternodes fit in the console? I guess we could just give it a try and see what happens :)